cpp/src/arrow/array/array_primitive.h

```diff
@@ -141,6 +147,8 @@ class ARROW_EXPORT DayTimeIntervalArray : public PrimitiveArray {
   // For compatibility with Take kernel.
   TypeClass::DayMilliseconds GetView(int64_t i) const { return GetValue(i); }
 
+  TypeClass::DayMilliseconds operator[](int64_t i) const { return Value(i); }
```

Comment:
Did you forget to update these ones as well?

Reply:
@pitrou These (time related ones) classes doesn't have Iterators defined. Am I missing something ?

Reply:
If they don't, then we should them.

Reply:
Okay 👍🏾